src/semver.nim

```diff
@@ -11,40 +11,53 @@ import std/strutils
 # it only handles basic dot-separated version format
 
 type Version* = ref object
-  major: int
-  minor: int
-  patch: int
-  name:  string
+  major:  int
+  minor:  int
+  patch:  int
+  suffix: string
+  name:   string
 
 proc parseVersion*(version: string): Version =
   var
-    major = 0
-    minor = 0
-    patch = 0
+    major  = 0
+    minor  = 0
+    patch  = 0
+    suffix = ""
   let
-    name  = version.strip(chars={'v', ',', '.'})
-    parts = name.split('.')
+    name     = version.strip(chars={'V', 'v'}, trailing=false).strip(chars={',', '.', '-', '+'})
+    sections = name.split({'-', '+'}, maxsplit=1)
+    parts    = sections[0].split('.')
   case len(parts):
     of 1:
-      major = parseInt(parts[0])
+      major  = parseInt(parts[0])
     of 2:
-      major = parseInt(parts[0])
-      minor = parseInt(parts[1])
+      major  = parseInt(parts[0])
+      minor  = parseInt(parts[1])
     of 3:
-      major = parseInt(parts[0])
-      minor = parseInt(parts[1])
-      patch = parseInt(parts[2])
+      major  = parseInt(parts[0])
+      minor  = parseInt(parts[1])
+      patch  = parseInt(parts[2])
     else:
       raise newException(ValueError, "Invalid or unsupported version format")
-  new result
-  result.name = name
-  result.major = major
-  result.minor = minor
-  result.patch = patch
+  if len(sections) == 2:
+    suffix = sections[1]
+  return Version(name:   name,
+                 major:  major,
+                 minor:  minor,
+                 patch:  patch,
+                 suffix: suffix)
 
 # version parts tuple used for comparison
-proc parts(self: Version): (int, int, int) =
-  return (self.major, self.minor, self.patch)
+proc parts(self: Version): (int, int, int, string) =
+  # TODO how to compare suffix?
+  # for now treating any suffix as less than no suffix
+  # assumping any suffix is for pre-releases which is not ideal
+  # correct but it is fine for chalk versions
+  # this handles things like 1-dev < 1.0
+  # no suffix is normalized to highest ascii char code \u7f
```

Comment:
Is there any risk to defaulting to a suffix of "\u7f" rather than an empty string "" for when there isn't a suffix present? Will this change break any calling code that would be surprised to get a "\uf7" value as a suffix?

Reply:
to clarify, if there is no suffix, the `\u7f` is added to both versions being compared so they will be equal. the difference comes into play only when one of the versions doesnt have the suffix

Reply:
Thanks for the additional clarity @miki725 and agree with you both that this seems more than good enough for now 👍